### PR TITLE
[Mac] Remove IdeInstanceConnection socket on Mac when running from bundle

### PR DESCRIPTION
**Pull request by Therzok**
Fixes VSTS #944831 - [FATAL] System.UnauthorizedAccessException exception in System.IO.File.Delete()
Fixes VSTS #938537 - [Watson] SIGABRT due to IdeInstanceConnection.StartListening

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeInstanceConnection.cs [hunk 1]
```diff
@@ -41,13 +40,17 @@ class IdeInstanceConnection
...
-		Socket listen_socket = null;
+		Socket listen_socket;
 		EndPoint ep;
 
 		public event EventHandler<FileEventArgs> FileOpenRequested;
 
-		public void Initialize (bool ipcTcp)
+		public IdeInstanceConnection (bool ipcTcp)
 		{
+			if (Platform.IsMac && Components.GtkWorkarounds.IsRunFromBundle ()) {
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeInstanceConnection.cs]: Please add a comment explaining why

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/GtkWorkarounds.cs [hunk 2]
```diff
@@ -1530,6 +1530,27 @@ public static void EmitRemoveSignal(Container container, Widget child)
...
 
+		internal static bool IsRunFromBundle ()
+		{
+			bool result = false;
+#if MAC
+			IntPtr mainBundle = GetMainBundle ();
+			var sel_bundleIdentifier = ObjCRuntime.Selector.GetHandle ("bundleIdentifier");
+
+			result = objc_msgSend_IntPtr (mainBundle, sel_bundleIdentifier) != IntPtr.Zero;
+			Console.WriteLine (result);
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/GtkWorkarounds.cs]: Debug code

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeInstanceConnection.cs (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeInstanceConnection.cs]: Fixed.

**Comment by Therzok** [comment]
@slluis do we want this in 8.2 or just 8.3?

**Comment by slluis** [comment]
Since it is low risk, let's merge to 8.2 too.

**Comment by slluis** [comment]
@monojenkins backport release-8.2

**Comment by slluis** [comment]
I mean 8.2.2

**Comment by baulig** [comment]
Is this related to / does this fix https://github.com/mono/mono/issues/15673?

**Comment by Therzok** [comment]
We should not see that anymore, but any other Socket.Bind can potentially trigger the mono bug

**Comment by Therzok** [comment]
@monojenkins backport release-8.2

**Pull request merged**